fs-storage/src/base_storage.rs

```diff
@@ -1,27 +1,37 @@
 use data_error::Result;
 use std::collections::BTreeMap;
 
+#[derive(Debug, PartialEq, PartialOrd, Ord, Eq, Clone)]
+/// Represents the synchronization status of the storage.
+pub enum SyncStatus {
+    /// No synchronization needed.
+    NoSync,
+    /// Load key-value mapping from the file system.
+    UpSync,
+    /// Write key-value mapping to the file system.
+    DownSync,
+    /// Changes from both in-memory and file system need to be merged before syncing.
+    FullSync,
+}
```

Comment:
Could you describe the motivation behind this separation? My guess is that `DownSync` is what should be used with cache storages, and any other storages seem to require `FullSync`? But when do we need `UpSync` or `NoSync`?